### PR TITLE
objectbox-generator: migrate to Conan v2

### PR DESCRIPTION
**Pull request by valgur**

### CODE REVIEW

**Review by RubenRBS — DISMISSED** [review]

File: recipes/objectbox-generator/all/test_package/conanfile.py [hunk 1]
```diff
@@ -1,11 +1,13 @@
-from conans import ConanFile, CMake, tools
-import os
+from conan import ConanFile
```
Comment (uilianries) [inline recipes/objectbox-generator/all/test_package/conanfile.py:1]:
```suggestion
from conan import ConanFile
from conan.tools.build import can_run
```

File: recipes/objectbox-generator/all/test_package/conanfile.py [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
+    test_type = "explicit"
+
+    def build_requirements(self):
+        self.tool_requires(self.tested_reference_str)
 
     def test(self):
-        if not tools.cross_building(self):
-            bin_path = os.path.join(self.deps_cpp_info["objectbox-generator"].rootpath, "bin", "objectbox-generator")
-            self.run(bin_path + " -help", run_environment=True)
+        self.run("objectbox-generator -help")
```
Comment (uilianries) [inline recipes/objectbox-generator/all/test_package/conanfile.py:13]:
```suggestion
        if can_run(self):
            self.run("objectbox-generator -help")
```
When cross-building, we will not be able to test it.

File: recipes/objectbox-generator/all/conanfile.py [hunk 3]
```diff
@@ -1,48 +1,60 @@
...
 
     def package_id(self):
         del self.info.settings.compiler
         del self.info.settings.build_type
 
+    def validate(self):
+        if self.settings.os not in ["Linux", "FreeBSD", "Windows", "Macos"] or self.settings.arch != "x86_64":
+            raise ConanInvalidConfiguration(f"{self.name} doesn't support current environment")
+
     def source(self):
```
Comment (uilianries) [inline recipes/objectbox-generator/all/conanfile.py:34]:
```suggestion
    def build(self):
```
When packaging pre-built binaries, the common senses is downloading to the build folder directly, it's can be removed after building as it does not contain sources: 

https://docs.conan.io/2/tutorial/creating_packages/other_types_of_packages/package_prebuilt_binaries.html#downloading-and-packaging-pre-built-binaries

File: recipes/objectbox-generator/all/conanfile.py [hunk 4]
```diff
@@ -1,48 +1,60 @@
...
-        tools.get(**self.conan_data["sources"][self.version][str(self.settings.os)],
-                  destination=self.source_folder)
-        tools.download(**self.conan_data["sources"][self.version]["License"], filename="LICENSE.txt")
+        get(self, **self.conan_data["sources"][self.version][str(self.info.settings.os)])
+        download(self, **self.conan_data["sources"][self.version]["License"], filename="LICENSE.txt")
+
+    def build(self):
+        pass
```
Comment (uilianries) [inline recipes/objectbox-generator/all/conanfile.py:39]:
```suggestion
```

File: recipes/objectbox-generator/all/conanfile.py [hunk 5]
```diff
@@ -1,48 +1,60 @@
...
     description = "ObjectBox Generator based on FlatBuffers schema files (fbs) for C and C++"
     license = "GPL-3.0"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/objectbox/objectbox-generator"
-    topics = ("database", "code-generator", "objectbox")
+    topics = ("database", "code-generator", "objectbox", "pre-built")
+
+    package_type = "application"
     settings = "os", "arch", "compiler", "build_type"
+    no_copy_source = True
```
Comment (uilianries) [inline recipes/objectbox-generator/all/conanfile.py:21]:
```suggestion
```
no_copy_source can be removed as binaries should be downloaded to the build folder directly, so there is nothing to copy from source to build folder.

**Review by uilianries — CHANGES_REQUESTED** [review]
Thank you for your contribution, some important consideration about packaging pre-built binaries.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by SSE4 — APPROVED** [review]

**Pull request merged**